src/controllers/user.controller.js

```diff
@@ -230,6 +230,56 @@ const getProfile = async (req, res) => {
   }
 }
 
+const createPost = async (req, res) => {
+  if (!req.user) {
+    return res.status(401).json({ message: 'Unauthorized.' })
+  }
+
+  try {
+    const {
+      rows: [post]
+    } = await executeQuery(
+      'INSERT INTO posts (user_id, text) VALUES ($1, $2) RETURNING *',
+      [req.user.id, req.body.text]
+    )
+
+    return res.status(200).json({ message: 'Successfully created post.', post })
+  } catch (error) {
+    if (error.code === '23502') {
+      return res.status(422).json({ message: `Missing field ${error.column}.` })
+    }
+
+    return res.status(500).json({ message: error })
+  }
+}
+
+const getPosts = async (req, res) => {
+  if (!req.user) {
+    return res.status(401).json({ message: 'Unauthorized.' })
+  }
+
+  const values = [req.user.id, req.body.limit || 5]
+  let str = ''
+  if (req.body.post_id) {
+    values.push(req.body.post_id)
+    str = 'AND posts.id < $3'
+  }
+
+  try {
+    const { rows } = await executeQuery(
+      `SELECT posts.*, first_name, last_name, email FROM posts
+      INNER JOIN users ON posts.user_id = users.id WHERE users.id = $1 ${str}
```

Comment:
We don't need to restrict it to the current user's posts. Remove that WHERE clause.